### PR TITLE
gambit-project: init at 16.1.1

### PR DESCRIPTION
**Pull request by t4ccer**
## Description of changes

Added [gambit project](http://www.gambit-project.org/) package

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ga/gambit-project/package.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  nativeBuildInputs = [ autoreconfHook ];
+
+  buildInputs = [ wxGTK31 ];
+
+  meta = with lib; {
+    description = "An open-source collection of tools for doing computation in game theory";
+    homepage = "http://www.gambit-project.org";
+    license = licenses.gpl2Only;
+    maintainers = with maintainers; [ t4ccer ];
+    platforms = platforms.linux ++ platforms.darwin ++ platforms.windows;
```
Comment (AndersonTorres) [inline pkgs/by-name/ga/gambit-project/package.nix:28]:
```suggestion
    platforms = platforms.unix ++ platforms.windows;
```

File: pkgs/by-name/ga/gambit-project/package.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    owner = "gambitproject";
+    repo = "gambit";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-ElPzJDQ1q+i1OyliychSUA9pT6yGSwjn/sKV0JX5wrQ=";
+  };
+
+  nativeBuildInputs = [ autoreconfHook ];
+
+  buildInputs = [ wxGTK31 ];
+
```
Comment (AndersonTorres) [inline pkgs/by-name/ga/gambit-project/package.nix:22]:
```suggestion

  strictDeps = true;

```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ga/gambit-project/package.nix (hunk 1 above)
Comment (t4ccer) [reply pkgs/by-name/ga/gambit-project/package.nix:28]: Why? The website doesn't mention BSD, etc. support

**Comment by t4ccer** [comment]
Actually `strictDeps = true;` broke `wx-config` detection, I'll revert it

File: pkgs/by-name/ga/gambit-project/package.nix (hunk 2 above)
Comment (t4ccer) [reply pkgs/by-name/ga/gambit-project/package.nix]: This breaks wx-config autconf detection

File: pkgs/by-name/ga/gambit-project/package.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/by-name/ga/gambit-project/package.nix:29]: Why not? The website does not mention other Unix-likes as unsupported.
You even included Darwin despite the error on Ofborg...

File: pkgs/by-name/ga/gambit-project/package.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  src = fetchFromGitHub {
+    owner = "gambitproject";
+    repo = "gambit";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-ElPzJDQ1q+i1OyliychSUA9pT6yGSwjn/sKV0JX5wrQ=";
+  };
+
+  nativeBuildInputs = [ autoreconfHook ];
+
+  buildInputs = lib.optional withGui wxGTK31;
```
Comment (AndersonTorres) [inline pkgs/by-name/ga/gambit-project/package.nix:22]:
```suggestion
  nativeBuildInputs = [ 
    autoreconfHook 
  ] ++ lib.optionals withGui [
    wxGTK31
  ];

  buildInputs = lib.optionals withGui [ 
    wxGTK31 
  ];
  
  strictDeps = true;
```

P.S.: is there any specific flag to set in order to build with GUI?

File: pkgs/by-name/ga/gambit-project/package.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{ stdenv
+, fetchFromGitHub
+, autoreconfHook
+, wxGTK31
+, lib
+, withGui ? true
+}:
```
Comment (AndersonTorres) [inline pkgs/by-name/ga/gambit-project/package.nix:7]: Nit:

```suggestion
{ lib
, autoreconfHook
, fetchFromGitHub
, stdenv
, wxGTK31
, withGui ? true
}:
```

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/by-name/ga/gambit-project/package.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  nativeBuildInputs = [ autoreconfHook ];
+
+  buildInputs = lib.optional withGui wxGTK31;
+
+  meta = with lib; {
+    description = "An open-source collection of tools for doing computation in game theory";
+    homepage = "http://www.gambit-project.org";
+    license = licenses.gpl2Only;
+    maintainers = with maintainers; [ t4ccer ];
+    platforms = platforms.linux ++ platforms.darwin ++ platforms.windows;
```
Comment (AndersonTorres) [inline pkgs/by-name/ga/gambit-project/package.nix:29]: Let's get rid of nested with:

```suggestion
  meta = {
    description = "An open-source collection of tools for doing computation in game theory";
    homepage = "http://www.gambit-project.org";
    license = lib.licenses.gpl2Only;
    maintainers = with lib.maintainers; [ t4ccer ];
    platforms = with lib.platforms; unix ++ windows;
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by t4ccer** [comment]
Alright, I got it working with `strictDeps`, trying to also figure out darwin failure but it's quite annoying without access to darwin machine

File: pkgs/by-name/ga/gambit-project/package.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/by-name/ga/gambit-project/package.nix:28]: Even when `wxGTK31` is in both `nativeBuildInputs` and `buildInputs`?

If this is the case, maybe it should be included at `depsBuildBuild` too...

Comment (AndersonTorres) [reply pkgs/by-name/ga/gambit-project/package.nix:28]: Also, does this software use pkg-config?

File: pkgs/by-name/ga/gambit-project/package.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    [ autoreconfHook ]
+    ++ lib.optional withGui wxGTK31;
+
+  buildInputs =
+    lib.optional withGui wxGTK31
+    ++ lib.optional stdenv.isDarwin darwin.apple_sdk.frameworks.Cocoa;
+
+  strictDeps = true;
+
+  configureFlags = lib.optional (!withGui) "--disable-gui";
```
Comment (AndersonTorres) [inline pkgs/by-name/ga/gambit-project/package.nix:31]:
```suggestion
  configureFlags = [
    (lib.enableFeature withGui "gui")
  ];
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Besides, LGTM.

**Review by AndersonTorres — APPROVED** [review]

**Comment by AndersonTorres** [comment]
@ofborg eval pkgsCross.mingwW64.gambit-project

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 282756` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gambit-project</li>
  </ul>
</details>

**Pull request merged**